### ISSUE TITLE
[Bug]: Cannot read properties of undefined (reading 'showOnlyCurrencies')

### ISSUE THREAD

**Issue by PeterYinusa** [MetaMask/metamask-extension#16080]
### Describe the bug

<img width="1594" alt="RC showOnlyCurrencies 15551" src="https://user-images.githubusercontent.com/53189696/193944775-395bc903-0d26-40ae-9e11-401c0068a3f1.png">

### Steps to reproduce

1. Login the extension
2. Switch to Goerli test network
3. Click the Buy button on the overview page

### Error messages or log output

```shell
Message: Cannot read properties of undefined (reading 'showOnlyCurrencies')

Code: TypeError

Stack:
TypeError: Cannot read properties of undefined (reading 'showOnlyCurrencies')
  at n.getIsBuyableMoonpayToken (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/common-8.js:1:250970)
  at chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-4.js:1:1056746
  at chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:205755
  at chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:206252
  at w (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-4.js:1:1056711)
  at ha (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:59903)
  at _u (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:104661)
  at El (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:90577)
  at gl (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:90502)
  at fl (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:87850)
  at chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:46308
  at n.unstable_runWithPriority (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:26:78615)
  at Ko (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:46017)
  at Zo (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:46253)
  at Jo (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:46188)
  at ue (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:114703)
  at Cn (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/ui-3.js:1:29663)
  at HTMLDocument.a (chrome-extension://noifopoklcadhlkenlkefaikmclmlild/sentry-install.js:1:11376)
```

### Version

10.21.0

### Browser

Chrome

### Operating system

MacOS

**Comment by PeterYinusa** [comment]
The fix here https://github.com/MetaMask/metamask-extension/pull/16031 needs to be pulled into the RC.
Lets leave this issue open until the bug is fixed on 10.21.0 RC for tracking

**Comment by seaona** [comment]
Closed by https://github.com/MetaMask/metamask-extension/pull/16031
